Change Status to be consistent with Link and Event (#1067)
open-telemetry/opentelemetry-specification: specification/trace/api.md

#### Set Status

Sets the `Status` of the `Span`. If used, this will override the default `Span`
status, which is `Unset`.

`Status` is semantically defined by the following properties:

- `StatusCanonicalCode` that represents the canonical set of `Status` codes.
- Optional `Description` that provides a descriptive message of the `Status`.

`StatusCanonicalCode` has the following values:

- `Unset`
  - The default status.
- `Ok`

>   - The operation has been validated by an Application developers or Operator to

Reviewer: ```suggestion
  - The operation has been validated by an Application developer or Operator to
```

  This API takes the `StatusCanonicalCode`, and optional `Description`,
  either as individual parameters or as an immutable object encapsulating them,
  whichever is most appropriate for the language.

The status code SHOULD remain unset, except for the following circumstances:

When the status is set to `ERROR` by Instrumentation Libraries, the status codes
SHOULD be documented and predictable. The status code should only be set to `ERROR`
according to the rules defined within the semantic conventions. For operations
not covered by the semantic conventions, Instrumentation Libraries SHOULD
publish their own conventions, including status codes.

Generally, Instrumentation Libraries SHOULD NOT set the status code to `Ok`,
unless explicitly configured to do so. Instrumention libraries SHOULD leave the
status code as `Unset` unless there is an error, as described above.

Application developers and Operators may set the status code to `Ok`.

Analysis tools SHOULD respond to an `Ok` status by suppressing any errors they
would otherwise generate. For example, to suppress noisy errors such as 404s.

Only the value of the last call will be recorded, and implementations are free

> to ignore previous calls.

Reviewer: This is only taken over but could be refined to be worded consistently (MUST/SHOULD/MAY) either here or in a follow-up.

#### Set Status

Sets the `Status` of the `Span`. If used, this will override the default `Span`

> status, which is `Unset`.

Author: This is fixing a bug in the specs (I guess)

`Status` is semantically defined by the following properties:

> - `StatusCanonicalCode` represents the canonical set of `Status` codes.

Reviewer 2: Why are we using such a long name as StatusCanonicalCode? Are there non-canonical status codes?

Author: Good question, I was not changing this, will file the issue #1068 to track this.

> - `StatusCanonicalCode` represents the canonical set of `Status` codes.
> - Description represents the descriptive message of the `Status`.

Reviewer 2: ```suggestion
- `StatusCanonicalCode` that represents the canonical set of `Status` codes.
- Optional `Description` that provides a descriptive message of the `Status`.
```

`StatusCanonicalCode` has the following values:

- `Unset`
  - The default status.
- `Ok`
  - The operation has been validated by an Application developers or Operator to

>     have completed successfully, or contain

Author: Paying the debt of https://github.com/open-telemetry/opentelemetry-specification/issues/1040

- `Error`
  - The operation contains an error.

The Span interface MUST provide:

> - An API to set the `Status` where the new `Status` properties are passed as
>   arguments. This SHOULD be called `SetStatus`. This API takes the
>   `StatusCanonicalCode`, and optional description.

Reviewer 2: ```suggestion
- An API to set the `Status`. This SHOULD be called `SetStatus`.
  This API takes the `StatusCanonicalCode`, and optional `Description`, 
  either as individual parameters or as an immutable object encapsulating them, 
  whichever is most appropriate for the language.
```

The status code SHOULD remain unset, except for the following circumstances:

When the status is set to `ERROR` by Instrumentation Libraries, the status codes
SHOULD be documented and predictable. The status code should only be set to `ERROR`
according to the rules defined within the semantic conventions. For operations
not covered by the semantic conventions, Instrumentation Libraries SHOULD
publish their own conventions, including status codes.

Generally, Instrumentation Libraries SHOULD NOT set the status code to `Ok`,
unless explicitly configured to do so. Instrumention libraries SHOULD leave the
status code as `Unset` unless there is an error, as described above.

Application developers and Operators may set the status code to `Ok`.

Analysis tools SHOULD respond to an `Ok` status by suppressing any errors they

> would otherwise generate. For example, to suppress noisy errors such as 404s.

Author: Good question, I was not changing this text, will file the issue https://github.com/open-telemetry/opentelemetry-specification/issues/1068 to track this.

Only the value of the last call will be recorded, and implementations are free
to ignore previous calls.

Note that there are multiple ways to implement this API, such as: overloading,

> variadic arguments, or define an immutable object to be passed to the API calls.

Reviewer 2: not necessary with the edit to the API description above.

Author: Correct, removed
